core/record/_record/recordNodeDependentsUpdater.js

```diff
@@ -7,25 +7,28 @@ import * as Node from '@core/record/node'
 import * as RecordExpressionParser from '@core/record/recordExpressionParser'
 import * as RecordExpressionValueConverter from '@core/record/recordExpressionValueConverter'
 import * as RecordReader from './recordReader'
-import * as RecordUpdater from './recordUpdater'
+import RecordUpdateResult from './RecordUpdateResult'
 
 const _logExpressionError = ({ error, expressionType, survey, nodeDef, expressionsToEvaluate, logger }) => {
   const surveyId = Survey.getId(survey)
   const nodeDefName = NodeDef.getName(nodeDef)
   const expressionsString = JSON.stringify(expressionsToEvaluate)
-  logger.error(
-    `error evaluating ${expressionType} in survey ${surveyId} for node def ${nodeDefName} / ${expressionsString}: ${error}`
-  )
+  const errorMessage = `error evaluating ${expressionType} in survey ${surveyId} for node def ${nodeDefName} / ${expressionsString}: ${error}`
+
+  if (logger?.errorEnabled) {
+    logger.error(errorMessage)
+  } else {
+    throw new Error(errorMessage)
+  }
 }
 
 /**
  * Module responsible for updating applicable and default values.
  */
 
-export const updateSelfAndDependentsApplicable = ({ survey, record, node, logger }) => {
+export const updateSelfAndDependentsApplicable = ({ survey, record, node, logger = null }) => {
   // Output
-  const nodesUpdatedToPersist = {}
-  const nodesWithApplicabilityUpdated = {}
+  const updateResult = new RecordUpdateResult({ record })
```

Comment:
updatedResult ?

Reply:
update result to me means "the result of an update"... updated means that it could have been updated later...

Reply:
resultOfUpdate

Reply:
do you mean the name of the variable or the name of the class?

Reply:
for example, mongodb has an UpdateResult class too, "The result of an update operation.": https://mongodb.github.io/mongo-csharp-driver/2.7/apidocs/html/T_MongoDB_Driver_UpdateResult.htm